ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -276,10 +282,19 @@ private boolean isRetriable(Throwable throwable) {
      * After TTL update finishes, call {@link SecurityService#postProcessRequest} to perform
      * request time security checks that rely on the request being fully parsed and any additional arguments set.
      * @param blobInfo the {@link BlobInfo} to use for security checks.
+     * @param blobId the {@link String} to use for blob id.
      * @return a {@link Callback} to be used with {@link Router#updateBlobTtl(String, String, long)}.
      */
-    private Callback<Void> routerTtlUpdateCallback(BlobInfo blobInfo) {
-      return buildCallback(frontendMetrics.updateBlobTtlRouterMetrics, blobId -> {
+    private Callback<Void> routerTtlUpdateCallback(BlobInfo blobInfo, String blobId) {
+      return buildCallback(frontendMetrics.updateBlobTtlRouterMetrics, convertedBlobId -> {
+        if (RestUtils.isUpsertForNamedBlob(restRequest.getArgs()) &&
```

Comment:
just make a helper since this pattern appears twice
```
boolean isUpsertForPermNamedBlob() {
 return RestUtils.isUpsertForNamedBlob(restRequest.getArgs()) &&
            blobInfo.getBlobProperties().getTimeToLiveInSeconds() == Utils.Infinite_Time;
}
```